tests/playwright/src/model/pages/volumes-page.ts

```diff
@@ -38,6 +38,30 @@ export class VolumesPage extends MainPage {
     this.collectUsageDataButton = this.additionalActions.getByRole('button', { name: 'Gather volume sizes' });
   }
 
+  public async getVolumeNameForContainer(containerName: string): Promise<string | undefined> {
+    try {
+      const rows = await this.getAllTableRows();
+      for (let i = rows.length - 1; i >= 0; i--) {
+        const volumeName = await rows[i].getByRole('cell').nth(3).getByRole('button').textContent();
+        if (volumeName) {
+          const volumeDetails = await this.openVolumeDetails(volumeName);
+          await volumeDetails.activateTab(VolumeDetailsPage.SUMMARY_TAB);
+          const volumeSummaryContent = await volumeDetails.tabContent.allTextContents();
+
+          for (const content of volumeSummaryContent) {
+            if (content.includes(containerName)) {
+              await volumeDetails.backLink.click();
+              return volumeName;
+            }
+          }
+          await volumeDetails.backLink.click();
+        }
+      }
+    } catch (err) {
+      console.log(`Exception caught on ${this.title} page with message: ${err}`);
+    }
+  }
+
```

Comment:
I'm honestly not sure what this method is doing, but it strikes me as a bad idea to do all those actions and navigations inside a for loop like that, can you please explain the flow of this method and why it should be like this?

Reply:
Of course! We need to find a volume that was added to the Kind cluster. However, the information indicating that the volume belongs to the Kind cluster is only in the summary tab of the volume details page. Therefore, I will go through all the volumes in the volumes table and check if the summary tab contains information about this volume being used by the Kind container.

Reply:
In that case that should be logic you implement inside the test, not inside a method, since this validation is test specific.

Reply:
You are trying to find an association of the container's volume in a volume Page. And to map it, you need to open volume's details summary page, If I am not mistaken. Unless we got that info in Volume Page rows directly, only way.

Reply:
Yes, I’m trying to find it on the details summary page and then just return the volume name to assert if it is running (used). Regarding implementing this method directly in the test, I think it could be useful in other tests as well. So, for me, the POM object is a good place for that functionality.

Reply:
Let's move it then outside of pom. into ie. utilities/operations.